client/apps/edit2/components/content/section_controls/test/index.test.js

```diff
@@ -1,188 +1,107 @@
+import configureStore from 'redux-mock-store'
 import React from 'react'
 import { mount } from 'enzyme'
-import Section from '/client/models/section.coffee'
+import { Provider } from 'react-redux'
+import { LayoutControls } from '../layout.jsx'
 import { SectionControls } from '../index'
 import { Fixtures } from '@artsy/reaction-force/dist/Components/Publishing'
-const { StandardArticle, FeatureArticle } = Fixtures
+const { StandardArticle } = Fixtures
 
 describe('Section Controls', () => {
   let props
 
+  SectionControls.prototype.isScrollingOver = jest.fn().mockReturnValue(true)
+  SectionControls.prototype.isScrolledPast = jest.fn().mockReturnValue(false)
+  $.fn.offset = jest.fn().mockReturnValue({top: 200})
+  $.fn.height = jest.fn().mockReturnValue(200)
+  $.fn.closest = jest.fn().mockReturnThis()
+
   const getWrapper = (props) => {
+    const mockStore = configureStore([])
+    const store = mockStore({
+      app: {
+        channel: props.channel
+      },
+      edit: {
+        article: props.article,
+        section: props.section,
+        sectionIndex: props.sectionIndex
+      }
+    })
+
```

Comment:
I think i've mentioned this before, but if you check out [here](https://github.com/artsy/force/blob/master/desktop/apps/auction/components/__tests__/Layout.test.js#L21) and [here](https://github.com/artsy/force/blob/master/desktop/apps/auction/__tests__/utils/renderTestComponent.js) I've written a redux test wrapper that you could adapt that makes most boilerplate go away. (It could probably be cleaned up some too):

API is: 

```javascript
const { wrapper } = renderTestComponent({
  Component: Layout,
  options: { renderMode: 'render' }, // from enzyme; shallow, render, mount
  // data is what is passed into the store before it is passed into the provider, and could  
  // be renamed to `initialState` for clarity
  data: {
    app: {
      auction: {
        name: 'An Auction'
      }
    }
  },
  // Props are whats passed into the component, and can potentially override 
  // data in initial state. 
  props: {
    ... 
  }
})
```

No pressure at all to use, just a suggestion. Since you're going down the redux route it would be good to set up a utility helper sooner rather than later that can hide some redundancy.

Reply:
Is true there are a number of places where this code is repeated -- will look into adding something to our test utils for these cases.  There are a few examples that require custom implementations -- for example in these tests we need to wrap the component in `<section>` for jQuery actions to work. So perhaps we'll end up with a mix of both.

Reply:
In terms of tests I don't necessarily think repetition is bad so either way 👍  whats more important is just keeping things literal, simple and clear, which current code definitely is.